djangocms_video/cms_plugins.py

```diff
@@ -1,61 +1,104 @@
 
 from cms.plugin_base import CMSPluginBase
 from cms.plugin_pool import plugin_pool
 
-from . import settings
-from .forms import VideoForm
-from .models import Video
+from . import models
 
 
-class VideoPlugin(CMSPluginBase):
-    model = Video
-    name = _("Video")
-    form = VideoForm
+class VideoPlayerPlugin(CMSPluginBase):
+    model = models.VideoPlayer
+    name = _('Video player')
     text_enabled = True
+    allow_children = True
+    child_classes = ['VideoSourcePlugin', 'VideoTrackPlugin']
 
-    render_template = "cms/plugins/video.html"
-
-    general_fields = [
-        ('movie', 'movie_url'),
-        'image',
-        ('width', 'height'),
-        'auto_play',
-        'auto_hide',
-        'fullscreen',
-        'loop',
-    ]
-    color_fields = [
-        'bgcolor',
-        'textcolor',
-        'seekbarcolor',
-        'seekbarbgcolor',
-        'loadingbarcolor',
-        'buttonoutcolor',
-        'buttonovercolor',
-        'buttonhighlightcolor',
+    fieldsets = [
+        (None, {
+            'fields': (
+                'template',
+                'label',
+            )
+        }),
+        (_('Embed video'), {
+            'classes': ('collapse',),
+            'fields': (
+                'embed_link',
+            )
+        }),
+        (_('Advanced settings'), {
+            'classes': ('collapse',),
+            'fields': (
+                'poster',
+                'attributes',
+            )
+        })
     ]
 
+    def render(self, context, instance, placeholder):
+        context = super(VideoPlayerPlugin, self).render(context, instance, placeholder)
+        context['video_template'] = instance.template
+        return context
+
+    def get_render_template(self, context, instance, placeholder):
+        return 'djangocms_video/{}/video_player.html'.format(instance.template)
+
+class VideoSourcePlugin(CMSPluginBase):
+    model = models.VideoSource
+    name = _('Source')
+    module = _('Video player')
+    require_parent = True
+    parent_classes = ['VideoPlayerPlugin']
+
     fieldsets = [
         (None, {
-            'fields': general_fields,
+            'fields': (
+                'source_file',
+                'text_title',
+            )
         }),
+        (_('Advanced settings'), {
+            'classes': ('collapse',),
+            'fields': (
+                'text_description',
+                'attributes',
+            )
+        })
     ]
-    if settings.VIDEO_PLUGIN_ENABLE_ADVANCED_SETTINGS:
-        fieldsets += [
-            (_('Color Settings'), {
-                'fields': color_fields,
-                'classes': ('collapse',),
-            }),
-        ]
 
-    def render(self, context, instance, placeholder):
-        context.update({
-            'object': instance,
-            'placeholder': placeholder,
+    def get_render_template(self, context, instance, placeholder):
+        return 'djangocms_video/{}/source.html'.format(context['video_template'])
```

Comment:
@Chive we did the same here on djangocms-audio. The recommendation from @czpython was actually to store it like that, to avoid problems with ``instance.parent``. Cause that was also my initial version.